add EBT Labels through CI

suchcodemuchwow opened the pull request:
This is a re-revert 🥹 for the [PR](https://github.com/elastic/kibana/pull/136843)

elasticmachine commented:
Pinging @elastic/kibana-operations (Team:Operations)

spalger commented:
A revert of the revert? Why was this reverted previously? What have you done to prevent that issue now?

suchcodemuchwow commented:
> A revert of the revert? Why was this reverted previously? What have you done to prevent that issue now?

After it got merged it started breaking in this [build](https://buildkite.com/elastic/kibana-on-merge/builds/19017#018244b7-f3bf-4893-b45c-8a3c796b32b7) and reverted [here](https://github.com/elastic/kibana/commit/371f091a14a53531948cbc96301a886e349d6e29). Don't have so much idea about why it just stopped working because it was green on PR build and when I tested locally. This one is also working locally and want to see the status in the PR now. I just added some [types](https://github.com/elastic/kibana/pull/137530/files#diff-35bdbcece3f4681b2f59ed2a105845187e2729fb2d0778208ec064acc4514863R27) and [JSON.stringify](https://github.com/elastic/kibana/pull/137530/files#diff-35bdbcece3f4681b2f59ed2a105845187e2729fb2d0778208ec064acc4514863R54) suggestion. @spalger

spalger commented on x-pack/test/performance/journeys/base.config.ts:
```diff
@@ -23,6 +24,18 @@ export default async function ({ readConfigFile, log }: FtrConfigProviderContext
+    ciBuildNumber: Number(process.env.BUILDKITE_BUILD_NUMBER) || 0,
+    gitRev: process.env.BUILDKITE_COMMIT,
+    isPr: !!process.env.GITHUB_PR_NUMBER,
+    prId: Number(process.env.GITHUB_PR_NUMBER) || 0,
```
We should just not specify this label when we're not in a PR
```suggestion
    ...(process.env.GITHUB_PR_NUMBER ? { prId: process.env.GITHUB_PR_NUMBER } : {}),
```

spalger commented on x-pack/test/performance/journeys/base.config.ts:
```diff
@@ -23,6 +24,18 @@ export default async function ({ readConfigFile, log }: FtrConfigProviderContext
 
   log.info(` 👷‍♀️ BUILD ID ${testBuildId}\n 👷 JOB ID ${testJobId}\n 👷‍♂️ EXECUTION ID:${executionId}`);
 
+  const telemetryLabels: TelemetryConfigLabels = {
+    branch: process.env.BUILDKITE_BRANCH,
+    ciBuildId: process.env.BUILDKITE_BUILD_ID,
+    ciBuildJobId: process.env.BUILDKITE_JOB_ID,
+    ciBuildNumber: Number(process.env.BUILDKITE_BUILD_NUMBER) || 0,
+    gitRev: process.env.BUILDKITE_COMMIT,
+    isPr: !!process.env.GITHUB_PR_NUMBER,
+    ...(process.env.GITHUB_PR_NUMBER ? { prId: process.env.GITHUB_PR_NUMBER } : {}),
+    testJobId,
+    testBuildId,
+  };
```
Sorry, didn't think about this but since this object is being JSON encoded this will set `prId` to a string, we should parse it to number

```suggestion
  const prId = process.env.GITHUB_PR_NUMBER ? Number.parseInt(process.env.GITHUB_PR_NUMBER, 10) : undefined;
  
  if (Number.isNaN(prId)) {
    throw new Error('invalid GITHUB_PR_NUMBER environment variable');
  }
  
  const telemetryLabels: TelemetryConfigLabels = {
    branch: process.env.BUILDKITE_BRANCH,
    ciBuildId: process.env.BUILDKITE_BUILD_ID,
    ciBuildJobId: process.env.BUILDKITE_JOB_ID,
    ciBuildNumber: Number(process.env.BUILDKITE_BUILD_NUMBER) || 0,
    gitRev: process.env.BUILDKITE_COMMIT,
    isPr: prId !== undefined,
    ...(prId !== undefined ? { prId } : {}),
    testJobId,
    testBuildId,
  };
```

suchcodemuchwow replied:
`...(process.env.GITHUB_PR_NUMBER ? { prId: Number(process.env.GITHUB_PR_NUMBER) } : {})` wouldn't this be sufficient change ? 
GITHUB_PR_NUMBER is either string or undefined:
- if it's undefined we will dest. empty object which is basically we are not gonna have this property
- if it's string then it means it's not empty and has some value "I' assume", I also assume that string will always a string-number

spalger replied:
Possibly, but passing values to the `Number()` constructor does insufficient parsing for different number formats and there's always a chance that `GITHUB_PR_NUMBER` is set to something we don't expect, maybe because someone tweaks the value or because something in Buildkite has changed. Adding parsing validation always seems like a good idea to me.

suchcodemuchwow replied:
Sure sounds good to me then 👍🏽

mistic reviewed (commented):
other than the changes suggestion to add parse validation, the rest lgtm
